src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -326,10 +340,14 @@ public static void Rethrow(object context, int opCode, int mdToken, long moduleV
             }
             catch (Exception ex)
             {
-                // profiled app will not continue working as expected without this method
-                var contextTypeName = context.GetType().FullName + " ";
-                var methodDef = $"{ResourceInvokerTypeName}.{nameof(Rethrow)}({contextTypeName} context)";
-                Log.ErrorException($"Error retrieving {methodDef}", ex);
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: ResourceInvokerTypeName,
+                    methodName: nameof(Rethrow),
+                    instanceType: context?.GetType().AssemblyQualifiedName);
```

Comment:
We're instrumenting a static method, so there's no instance in this case. `context` is the first argument.

Reply:
Same as above, and actually, especially important for this one.

Reply:
Looking at the source code for `ResourceInvoker`, there are 3 `Rethrow()` overloads, and neither one is an extension method. If we are instrumenting all 3, then maybe the log message should include the method signature, since they are really 3 different methods:
```csharp
ResourceInvoker.Rethrow(ResourceExecutedContextSealed);
ResourceInvoker.Rethrow(ExceptionContextSealed);
ResourceInvoker.Rethrow(ResultExecutedContextSealed);
````

This raises another question: if we're ignoring the argument type when matching the `Rethrow()` method, are we caching the generated IL correctly? We need to generate and cache all 3 versions since each one calls into a different target method.

Reply:
It's cached by mdToken as well as the parameter types currently, which is plenty to differentiate.

Reply:
The differentiator is the first parameter, which is why I thought it was valuable.